server/lib/taxForms.js

```diff
@@ -0,0 +1,88 @@
+import models from '../models';
+import { uniqBy } from 'lodash';
+const { RequiredLegalDocument, LegalDocument, Collective, User } = models;
+const {
+  documentType: { US_TAX_FORM },
+} = RequiredLegalDocument;
+
+export async function findUsersThatNeedToBeSentTaxForm({ invoiceTotalThreshold, year }) {
+  const users = await RequiredLegalDocument.findAll({
+    where: { documentType: US_TAX_FORM },
+  })
+    .map(requiredUsTaxDoc => requiredUsTaxDoc.getHostCollective())
+    .map(host => host.getUsersWhoHaveTotalExpensesOverThreshold({ threshold: invoiceTotalThreshold, year }))
+    .reduce((acc, item) => {
+      return [...acc, ...item];
+    }, [])
+    .filter(user => {
+      return LegalDocument.doesUserNeedToBeSentDocument({ documentType: US_TAX_FORM, year, user });
+    });
+
+  return uniqBy(users, 'id');
+}
+
+export async function isUserTaxFormRequiredBeforePayment({ invoiceTotalThreshold, year, HostCollectiveId, UserId }) {
+  const host = await Collective.findByPk(HostCollectiveId);
+  const user = await User.findByPk(UserId);
+  const requiredDocuments = await host.getRequiredLegalDocuments({
+    where: {
+      documentType: US_TAX_FORM,
+    },
+  });
+
+  const isOverThreshold = await host.doesUserHaveTotalExpensesOverThreshold({
+    threshold: invoiceTotalThreshold,
+    year,
+    UserId,
+  });
+  const hasUserCompletedDocument = await LegalDocument.hasUserCompletedDocument({
+    documentType: US_TAX_FORM,
+    year,
+    user,
+  });
+
+  return requiredDocuments.length > 0 && isOverThreshold && !hasUserCompletedDocument;
```

Comment:
Maybe we can avoid additional queries by checking things just after the queries?

```es6
const requiredDocuments = await host.getRequiredLegalDocuments({ ... })
if (!requiredDocuments.length) {
  return false;
}
```

Reply:
Yeah, cool. Done.